Greenshot/Forms/ImageEditorForm.cs

```diff
@@ -1444,28 +1463,129 @@ private void TornEdgesToolStripMenuItemMouseUp(object sender, MouseEventArgs e)
 		}
 
 		private void ImageEditorFormResize(object sender, EventArgs e) {
+			AlignCanvasPositionAfterResize();
+		}
+
+		private void AlignCanvasPositionAfterResize() {
 			if (Surface?.Image == null || panel1 == null) {
 				return;
 			}
-			Size imageSize = Surface.Image.Size;
-			Size currentClientSize = panel1.ClientSize;
 			var canvas = Surface as Control;
+			Size canvasSize = canvas.Size;
+			Size currentClientSize = panel1.ClientSize;
 			Panel panel = (Panel) canvas?.Parent;
 			if (panel == null) {
 				return;
 			}
 			int offsetX = -panel.HorizontalScroll.Value;
 			int offsetY = -panel.VerticalScroll.Value;
-			if (currentClientSize.Width > imageSize.Width) {
-				canvas.Left = offsetX + (currentClientSize.Width - imageSize.Width) / 2;
+			if (currentClientSize.Width > canvasSize.Width) {
+				canvas.Left = offsetX + (currentClientSize.Width - canvasSize.Width) / 2;
 			} else {
 				canvas.Left = offsetX + 0;
 			}
-			if (currentClientSize.Height > imageSize.Height) {
-				canvas.Top = offsetY + (currentClientSize.Height - imageSize.Height) / 2;
+			if (currentClientSize.Height > canvasSize.Height) {
+				canvas.Top = offsetY + (currentClientSize.Height - canvasSize.Height) / 2;
 			} else {
 				canvas.Top = offsetY + 0;
 			}
 		}
+
+		/// <summary>
+		/// Compute a size as a sum of surface size and chrome.
+		/// Upper bound is working area of the screen. Lower bound is fixed value.
+		/// </summary>
+		private Size GetOptimalWindowSize() {
+			var surfaceSize = (Surface as Control).Size;
+			var chromeSize = GetChromeSize();
+			var newWidth = chromeSize.Width + surfaceSize.Width;
+			var newHeight = chromeSize.Height + surfaceSize.Height;
+
+			// Upper bound. Don't make it bigger than the available working area.
+			var maxWindowSize = GetAvailableScreenSpace();
+			newWidth = Math.Min(newWidth, maxWindowSize.Width);
+			newHeight = Math.Min(newHeight, maxWindowSize.Height);
+
+			// Lower bound. Don't make it smaller than a fixed value.
+			int minimumFormWidth = 650;
+			int minimumFormHeight = 530;
+			newWidth = Math.Max(minimumFormWidth, newWidth);
+			newHeight = Math.Max(minimumFormHeight, newHeight);
+
+			return new Size(newWidth, newHeight);
+		}
+
+		private Size GetChromeSize()
+			=> Size - panel1.ClientSize;
+
+		/// <summary>
+		/// Compute a size that the form can take without getting out of working area of the screen.
+		/// </summary>
+		private Size GetAvailableScreenSpace() {
+			var screen = Screen.FromControl(this);
+			var screenBounds = screen.Bounds;
+			var workingArea = screen.WorkingArea;
+			if (Left > screenBounds.Left && Top > screenBounds.Top) {
+				return new Size(workingArea.Width - Left, workingArea.Height - Top);

```

Comment:
```suggestion
				return new Size(workingArea.Right - Left, workingArea.Bottom - Top);
```
Fixes the [secondary monitor issue](https://github.com/greenshot/greenshot/pull/201#issuecomment-619582140).
For the primary monitor e.g. `Width` and `Right` are the same, but not for the secondary.

Reply:
Ah right. I messed up concrete screen and continuous coordinate space where it also lies itself. Thanks